app/services/order_factory.rb

```diff
@@ -65,6 +79,10 @@ def set_addresses
     @order.update_attributes(attrs.slice(:bill_address_attributes, :ship_address_attributes))
   end
 
+  def set_shipping_method
+    @order.update_attributes!(shipping_method_id: attrs[:shipping_method_id])
```

Comment:
the column order.shipping_method_id is deprecated and will be deleted after v2 is live. we cannot rely on it.
this is how you do it in v2:
@order.shipment.select_shipping_method(attrs[:shipping_method_id])